### DISCUSSION TITLE
Step 2 not updating README to Step 3

### DISCUSSION THREAD

**Discussion by rluna319** [skills/.github#324] [Release-based workflow] [answered]
Hello, 

I am having an issue where upon completing the instructions in the README for Step 2 of the Release-based workflow tutorial, the README does not update and move to Step 3.

I didn't spot this discussion board link till after I had posted an [issue on the official relelase-based-workflow repo](https://github.com/skills/release-based-workflow/issues/50#issue-1847291398). I tagged the issue as a bug and I put the details there.

**Comment by jmbvstepback** [comment]
Hi, the same here. Non updating to Step 3 and completely blocked in the progress on the associated course :-( . Any help is welcome....

**Reply by jmbvstepback** [reply]
Update: In my case, updating the case was enough to pass from Step2 to Step3. Check the name of the function is agree in case with:

```
function calculateNumbers
```

**Reply by sinsukehlab** [reply]
Your issue is on the "Code with Copilot" course. It's different from the issue that the original post discusses.

**Comment by sinsukehlab** [comment, marked as answer]
According to the network graph, it seems that you didn't create your new branches from the `main` branch.
The Step 2 workflow ran on the `release-v1.0` branch and skipped the update job because the Step was not 2 but 0 on the `release-v1.0` branch.

**Reply by rluna319** [reply]
I tried again creating a new branch off the main. After waiting and refreshing nothing happened. I went to the actions tab and re-ran the jobs. It still showed it skipping. I am not sure if i need to do something else or if maybe its still running the job based on my incorrect branch. So, a couple more questions.

1. How do you view that network graph?
2. How do I proceed?

I appreciate the help. Thank you.

**Reply by sinsukehlab** [reply]
The Step 2 workflow didn't run at all (not just skipping). It's probably because merging the PR rluna319/skills-release-based-workflow#3 didn't push a commit to that "base.css" file path of the `release-v1.0` branch, which had already been edited in the PR rluna319/skills-release-based-workflow#2.
1. The network graph is on the Insights tab.
1. Actually, you have edited the wrong section of the "base.css" file. There is a comment indicating the section that requires editing. You can now proceed by fixing the "base.css" file of the `release-v1.0` branch (another PR is recommended).

**Reply by rluna319** [reply]
Awesome! I'm still a little new to CSS so I didn't realize that I was modifying what I assume is the declaration of the 'body' selector. But I have fixed the issues on a new branch, created another PR, and saw that both jobs ran this time. The README is now updated to Step 3. 

Thank you so much for you help! I'll go ahead and close that issue I created on the original repo.

**Reply by sinsukehlab** [reply]
I'm glad that I could be of help.

**Reply by zzzbra** [reply]
This isn't really an answer since the directions provided specifically instruct one to merge to `release-v1.0`, not `main`. Either the workflow should be updated or the instructions provided in the Step 2 README should be updated.

**Reply by sinsukehlab** [reply]
I haven't instructed the author of this question to merge to the `main` branch. According to the Step 2 markdown, one should create a new branch off of the `main` branch, edit `base.css`, and open and merge a PR with `release-v1.0` as the base and the new branch as the compare.

**Reply by zzzbra** [reply]
I don't think that's accurate:
> Open a pull request with release-v1.0 as the base branch, and your new branch as the compare branch.

What then gets merged to in the subsequent commands is the release branch. If there's no update to `main`, the workflow won't trigger, unless I am mistaken.

**Reply by sinsukehlab** [reply]
You made the same mistake as the questioner. You should have created your new branch from the `main` branch, not from the `release-v1.0` branch.
The workflow on the `release-v1.0` branch will be triggered. Actually, you have also triggered the workflow. The step won't be updated unless the step on the `release-v1.0` branch is 2 after merging.

**Reply by zzzbra** [reply]
The assumption that the original change be checked out off of `main` is not implied anywhere, and is pretty odd. If one has a release branch they are prepping, and knows a change is meant for it, they are not likely to check out off of the divergent `main` branch. Either explicitly state this assumption in the instructions, or fix the logic of the workflow updates to match the more real world workflow, or folks are liable to continue adding to this thread.

**Reply by sinsukehlab** [reply]
The assumption is explicitly stated in the instructions.

### :keyboard: Activity: Update `base.css`

1. Create a new branch off of the `main` branch and change the `body` CSS declaration in `base.css` to match what is below. This will set the page background to black.

```
body {
    background-color: black;
}
```

1. Open a pull request with `release-v1.0` as the `base` branch, and your new branch as the `compare` branch.
1. Fill in the pull request template to describe your changes.
1. Click **Create pull request**.

**Comment by cechabuilt** [comment]
I had a problem where I wasn't able to go to step 3 > 4.  

I could not figure it out.  Ended up deleting the entire repository and redoing all the steps exactly the same.  

It ended up working the second time around.   

IDK if this will help you  but maybe give that a try?

**Reply by rluna319** [reply]
That might have worked too if I followed the instructions properly in the new repo. I wanted to learn a bit more by troubleshooting the issue without deleting and forking the repo again.

I solved the issue thanks to @sinsukehlab. You can view the process in [the answer thread](https://github.com/orgs/skills/discussions/324#discussioncomment-6717583). But thank you for your contribution to my issue!

**Reply by cechabuilt** [reply]
Glad you were able to figure it out!!

**Comment by HakaCode** [comment]
Hello, I would like some help if possible...
I have an error when automatically updating the readme, after **Step 7: Create release v1.0.1**

**Reply by sinsukehlab** [reply]
I can't find any mistakes in your repo. Try triggering the Step 7 workflow manually (workflow_dispatch).
